src/lock_file/satisfiability/mod.rs

```diff
@@ -1471,10 +1478,30 @@ pub(crate) async fn verify_package_platform_satisfiability(
             ))
         })?;
 
+        let discovered_backend = DiscoveredBackend::discover(
+            &source_dir,
+            &environment.channel_config(),
+            &EnabledProtocols::default(),
+        )
+        .map_err(PlatformUnsat::BackendDiscovery)
+        .map_err(Box::new)?;
+
+        let project_model_hash =
```

Comment:
Lets make a function out of this that both satisiabilty and the creation share to avoid drift.